clib/cUnix.ml

```diff
@@ -69,7 +69,7 @@ let canonical_path_name p =
     p'
   with Sys_error _ ->
     (* We give up to find a canonical name and just simplify it... *)
-    strip_path p
+    current ^ dirsep ^ strip_path p
```

Comment:
Are we never called with an absolute path that doesn't start with PWD? Like `/doesnotexist` or some such thing.
Also stripping a leading PWD just to put it back is kinda strange.

Reply:
Fair enough, but it is hard to tell what a function is supposed to do when it is not documented.

If you want to add a test for a leading / you are welcome to do so. Note that strip_path also gets rid of stuff like `/./`, so you should not just return `p` IMO.